libs/server/Resp/AsyncProcessor.cs

```diff
@@ -0,0 +1,138 @@
+            where TGarnetApi : IGarnetApi
+        {
+            unsafe
+            {
+                while (!RespWriteUtils.WriteError($"ASYNC {asyncStarted}", ref dcurr, dend))
+                    SendAndReset();
+            }
+
+            if (++asyncStarted == 1) // first async operation on the session, create the IO continuation processor
+            {
+                asyncWaiter = new()
+                {
+                    RunContinuationsAsynchronously = true
+                };
+                var _storageApi = storageApi;
+                _ = Task.Run(async () => await AsyncGetProcessor(_storageApi));
+            }
+            else
+            {
+                Debug.Assert(asyncWaiter != null);
+                asyncWaiter.Signal();
+            }
+        }
+
+        /// <summary>
+        /// Background processor for async IO continuations. This is created only when async is turned on for the session.
+        /// It handles all the IO completions and takes over the network sender to send async responses when ready.
+        /// Note that async responses are not guaranteed to be in the same order that they are issued.
+        /// </summary>
+        async Task AsyncGetProcessor<TGarnetApi>(TGarnetApi storageApi)
+            where TGarnetApi : IGarnetApi
+        {
+            while (true)
+            {
+                while (asyncCompleted < asyncStarted)
+                {
+                    // First complete all pending ops
+                    storageApi.GET_CompletePending(out var completedOutputs, true);
+
+                    try
+                    {
+                        unsafe
+                        {
+                            // We are ready to send responses so we take over the network sender from the main ProcessMessage thread
+                            // Note that we cannot take over while ProcessMessage is in progress because partial responses may have been
+                            // sent at that point. For sync responses that span multiple ProcessMessage calls (e.g., MGET), we need the
+                            // main thread to hold the sender lock until the response is done.
+                            networkSender.EnterAndGetResponseObject(out dcurr, out dend);
+
+                            // Send async replies with completed outputs
+                            while (completedOutputs.Next())
+                            {
+                                // This is the only thread that updates asyncCompleted so we do not need atomics here
+                                asyncCompleted++;
+                                var o = completedOutputs.Current.Output;
+
+                                // We write async push response as an array: [ "async", "<token_id>", "<result_string>" ]
+                                RespWriteUtils.WritePushLength(3, ref dcurr, dend);
+                                RespWriteUtils.WriteBulkString("async"u8, ref dcurr, dend);
```

Comment:
nit: CmdStrings.ASYNC

Reply:
Added `CmdStrings.async` for the lowercase version and used it.